tests/ui/thread-local/main-thread-dtor.rs

```diff
@@ -0,0 +1,31 @@
+//@ run-pass
+//@ check-run-results
+//@ needs-threads (really only needs TLS, not threads, but those seem to usually come together)
```

Comment:
I'm pretty sure thread-local destructors never run on platforms where threads are not supported even if they support `thread_local!()`. On those platforms we implement thread-locals using a regular static.

Reply:
Ah right, I meant to update the comment.